### ISSUE TITLE
[SUGGESTION] Avoidance of compiler warning

### ISSUE THREAD

**Issue by drmason789** [cagnulein/qdomyos-zwift#936]
I see this one coming up a lot in the build output:

```
In file included from debug\../../../qdomyos-zwift-911-testing/src/dirconmanager.h:15,
                 from debug\../../../qdomyos-zwift-911-testing/src/virtualbike.h:29,
                 from debug\../../../qdomyos-zwift-911-testing/src/domyostreadmill.h:31,
                 from debug\../../../qdomyos-zwift-911-testing/src/mainwindow.h:4,
                 from debug\../../../qdomyos-zwift-911-testing/src/charts.h:4,
                 from debug\moc_charts.cpp:10:
debug\../../../qdomyos-zwift-911-testing/src/dirconprocessor.h: In copy constructor 'DirconProcessorCharacteristic::DirconProcessorCharacteristic(const DirconProcessorCharacteristic&)':
debug\../../../qdomyos-zwift-911-testing/src/dirconprocessor.h:21:5: warning: base class 'class QObject' should be explicitly initialized in the copy constructor [-Wextra]
     DirconProcessorCharacteristic(const DirconProcessorCharacteristic &cp) { this->operator=(cp); }

```

Notably:

debug\../../../qdomyos-zwift-911-testing/src/dirconprocessor.h: In copy constructor 'DirconProcessorCharacteristic::DirconProcessorCharacteristic(const DirconProcessorCharacteristic&)':
debug\../../../qdomyos-zwift-911-testing/src/dirconprocessor.h:21:5: warning: **base class 'class QObject' should be explicitly initialized in the copy constructor [-Wextra]**
     DirconProcessorCharacteristic(const DirconProcessorCharacteristic &cp) { this->operator=(cp); }

I don't think this class is supposed to have a copy constructor:

See: [QObject Class](https://doc.qt.io/qt-5/qobject.html)

![image](https://user-images.githubusercontent.com/29954900/189540606-1f7d3314-c93f-496f-afa9-90a0d8675bbd.png)

"_QObject has neither a copy constructor nor an assignment operator. This is by design. Actually, they are declared, but in a private section with the macro [Q_DISABLE_COPY](https://doc.qt.io/qt-5/qobject.html#Q_DISABLE_COPY)(). In fact, all Qt classes derived from QObject (direct or indirect) use this macro to declare their copy constructor and assignment operator to be private. The reasoning is found in the discussion on [Identity vs Value](https://doc.qt.io/qt-5/object.html#identity-vs-value) on the Qt [Object Model](https://doc.qt.io/qt-5/object.html) page._"

The app still builds with this constructor removed, but I'm not sure why it's there and I'm not in a position to test it.

**Comment by cagnulein** [comment]
Yes I have noticed it too, but I was too lazy to remove the warning since I would have to test again the dircon :)

**Comment by cagnulein** [comment]
no leave it as open, i have just to find the time to test it again :)

**Comment by cagnulein** [comment]
tested and it works flawlessy! thanks @drmason789

**Comment by cagnulein** [comment]
another good good thing would be remove ALL the warnings from the code. Must of them are from the FIT SDK, maybe there is a way to ignore all the warning from a subfolder @drmason789  ?

**Comment by drmason789** [comment]
Perhaps make the fit-sdk a separate project compiling to a static lib and use different warning options for it.

**Comment by cagnulein** [comment]
mmm but I have to build it for all the architecture...I have to think about this..It will be better to specify a different warning level, or trying to fix the warnings
